src/app/data-model-provider/Provider.h

```diff
@@ -47,6 +47,9 @@ class Provider : public ProviderMetadataTree
 public:
     virtual ~Provider() = default;
 
+    // Initialize the data model internal code to be ready to send and receive data model messages.
+    virtual void InitDataModel() = 0;
```

Comment:
Why do we have two separate init things (Startup and this)?

It feels like we are basically flailing here instead of trying to actually decide what our startup system should look like....

What it _should_ look like, imo, is that we don't start up data model anything until we have started core stack services (fabric table, etc).

That is, the call to Startup() on this thing should be where InitDataModelHandler() is now, no?  This was provided as review feedback on earlier PRs, and pushed off into followups, but this is that followup and now we are doing weird things.

Reply:
The Startup function is invoked by both the controller and server. However, Ember-specific InitDataModelHandler initialization is only required on the server side. If this initialization is included in the Startup function, it will cause numerous controller unit tests to fail.

Reply:
> However, Ember-specific InitDataModelHandler initialization is only required on the server side.

Controller calls InitDataModelHandler in CHIPDeviceControllerFactory.cpp.  That call should go away, and both codepaths should have InitDataModelHandler happening entirely within the data model provider, no?

Reply:
If I move InitDataModelHandler into the Startup function as you suggested, it will alter the current initialization logic. This change causes InitDataModelHandler to be called by TestServerCommandDispatch, leading to multiple test failures. I've spent two days debugging this issue without making progress.

I believe the goal of this PR is to decouple InitDataModelHandler from libCHIP while maintaining the existing initialization logic and ensuring all unit tests pass as they currently do, correct?

Reply:
Per offline discuss with Andrie - to temporary patch the TestServerCommandDispatch we could override Startup function to have a Startup that does not do InitDataModelHandler.